driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -183,4 +185,41 @@ public T reduce(final T initialValue, final BiFunction<T,  T,  T> in) {
                 .toBsonDocument(BsonDocument.class, cr)).apply(cr));
     }
 
+
+    /** @see StringExpression */
+
+    @Override
+    public StringExpression toLower() {
+        return new MqlExpression<>(ast("$toLower"));
+    }
+
+    @Override
+    public StringExpression toUpper() {
+        return new MqlExpression<>(ast("$toUpper"));
+    }
+
+    @Override
+    public StringExpression concat(final StringExpression concat) {
+        return new MqlExpression<>(ast("$concat", concat));
+    }
```

Comment:
Can you provide an example of doing that?  It may be planned for the javadoc. 

I just want to make sure users coming from the [$concat docs](https://www.mongodb.com/docs/manual/reference/operator/aggregation/concat/) will easily be able to translate to this api.

Reply:
`abc = ofStringArray("a", "b", "c")`

Presently, they'd need to write `abc.reduce(of(""), (a, b) -> a.concat(b));`

It would be preferable to write `abc.concat()`, but that syntax is elusive, because we don't have extension methods as in C#. I discuss the problem and give an example here https://github.com/mongodb/mongo-java-driver/pull/1031#discussion_r1013433929 using "add" rather than "concat".

Reply:
Can we document this scenario (concatenating multiple strings) and the solution on the `concat` method?

Reply:
If documented on concat, it should be documented on sum, add, or (any), and (all), min, max, as well as all other reductive operations, of which there are many. It may be best to document this in some general location.

Let's defer this until writing docs. I have added a comment to the final expressions MR: https://github.com/mongodb/mongo-java-driver/pull/1032#discussion_r1021685532